### ISSUE TITLE
Invalid Date object shown in JSON response

### ISSUE THREAD

**Issue by schrenkb1** [vpulim/node-soap#705]
I am currently calling a WSDL describe and get something like this.
{ input: { 'Value[]': 'xsd:string' },
  output: 
   { 'ItemResponse[]': 
      { ItemEffecDate: 'xsd:date',
        ItemExpDate: 'xsd:date',
        ItemStatus: 'xsd:string',
        ItemTotal: 'xsd:decimal'} } }

So I send something like this.
{ Value[]: ['456 ASD', '987 WER'] }

My response comes back as 
   { ItemResponse: 
    [ { ItemEffecDate: Invalid Date,
        ItemExpDate: Invalid Date,
        ItemStatus: 'Full Time',
        ItemTotal: '456.45'},
      { ItemEffecDate: Invalid Date,
        ItemExpDate: Invalid Date,
        ItemStatus: 'Full Time',
        ItemTotal: '456.45'}] } }

The original XML response shows a date as:
2012-12-03-05:00 

I'm not sure where the conversion here is going wrong any ideas?

**Comment by schrenkb1** [comment]
I fixed this problem for myself but will leave it here for others if they have an issue. 

On line 1356 - 1357 of wsdl.js there is the code 
    } else if (name === 'dateTime' || name === 'date') {
    value = new Date(text);
This was trying to convert a item that was listed as a date into a date but javascript was not recognizing it as a date that it recognized therefore stores a null object and lists the text Invalid Date. 

How did I fix this for me...
Well I edited the file to allow the date item to just store as text value and am dealing with the rest on the front end of my application.

Code now looks like:
     } else if (name === 'dateTime' || name === 'date') {
      value = text;
